lib/processors/minifier.js

```diff
@@ -40,57 +63,56 @@ const debugFileRegex = /((?:\.view|\.fragment|\.controller|\.designtime|\.suppor
- * 				Whether to add a sourceMappingURL reference to the end of the minified resource
+ *   Whether to add a sourceMappingURL reference to the end of the minified resource
+ * @param {boolean} [parameters.options.useWorkers=false]
+ *  Whether to offload the minification task onto separate CPU threads. This often speeds up the build process
  * @returns {Promise<module:@ui5/builder/processors/minifier~MinifierResult[]>}
- * 				Promise resolving with object of resource, dbgResource and sourceMap
+ *   Promise resolving with object of resource, dbgResource and sourceMap
  */
-export default async function({resources, options: {addSourceMappingUrl = true} = {}}) {
+export default async function({resources, taskUtil, options: {addSourceMappingUrl = true, useWorkers = false} = {}}) {
+	let minify;
+	if (useWorkers) {
+		if (!taskUtil) {
+			// TaskUtil is required for worker support
+			throw new Error(`Minifier: Option 'useWorkers' requires a taskUtil instance to be provided`);
+		}
+		minify = minifyInWorker;
+	} else {
+		// Do not use workerpool
+		minify = (await import("./minifierWorker.js")).default;
+	}
+
 	return Promise.all(resources.map(async (resource) => {
 		const dbgPath = resource.getPath().replace(debugFileRegex, "-dbg$1");
 		const dbgResource = await resource.clone();
 		dbgResource.setPath(dbgPath);
 
 		const filename = posixPath.basename(resource.getPath());
 		const code = await resource.getString();
-		try {
-			const sourceMapOptions = {
-				filename
-			};
-			if (addSourceMappingUrl) {
-				sourceMapOptions.url = filename + ".map";
-			}
-			const dbgFilename = posixPath.basename(dbgPath);
-			const result = await minify({
-				// Use debug-name since this will be referenced in the source map "sources"
-				[dbgFilename]: code
-			}, {
-				output: {
-					comments: copyrightCommentsAndBundleCommentPattern,
-					wrap_func_args: false
-				},
-				compress: false,
-				mangle: {
-					reserved: [
-						"jQuery",
-						"jquery",
-						"sap",
-					]
-				},
-				sourceMap: sourceMapOptions
-			});
-			resource.setString(result.code);
-			const sourceMapResource = new Resource({
-				path: resource.getPath() + ".map",
-				string: result.map
-			});
-			return {resource, dbgResource, sourceMapResource};
-		} catch (err) {
-			// Note: err.filename contains the debug-name
-			throw new Error(
-				`Minification failed with error: ${err.message} in file ${filename} ` +
-				`(line ${err.line}, col ${err.col}, pos ${err.pos})`);
+
+		const sourceMapOptions = {
+			filename
+		};
+		if (addSourceMappingUrl) {
+			sourceMapOptions.url = filename + ".map";
 		}
+		const dbgFilename = posixPath.basename(dbgPath);
+
+		const result = await minify({
+			filename,
+			dbgFilename,
+			code,
+			sourceMapOptions
+		}, taskUtil);
+		resource.setString(result.code);
+		const sourceMapResource = new Resource({
+			path: resource.getPath() + ".map",
+			string: result.map
+		});
+		return {resource, dbgResource, sourceMapResource};
```

Comment:
`await` here blocks the code flow. 
Until minify in workerpool is finished, all other minification tasks are waiting. You could verify this by checking the `getPool(taskUtil).stats()` at this point.

An unblocking option might be to stash the promise of `minify(..)` so that the rest of the `resources.map` are queued to the pool and then wait for the minification inside the promise.
Something like this:

```suggestion
// Clock defined at the top of the file
import {performance} from "perf_hooks";
function clock(start) {
    if (!start) return performance.now();
	return performance.now() - start;
}

// ....

		const start = clock();
		const resultP = minify({
			filename,
			dbgFilename,
			code,
			sourceMapOptions
		}, taskUtil);

		return resultP.then(function(result) {
			console.log(`[${filename}] workerpool minification 
				took ${clock(start)}ms ${JSON.stringify(getPool(taskUtil).stats())}`);

			resource.setString(result.code);
			const sourceMapResource = new Resource({
				path: resource.getPath() + ".map",
				string: result.map
			});
			return {resource, dbgResource, sourceMapResource};
		});
```